pkg/route/graph/analysis/analysis.go

```diff
@@ -70,6 +76,56 @@ route:
 
 				continue route
 			}
+
+			if routeNode.Spec.Port == nil {
+				continue route
```

Comment:
We've got too many of these now.  Refactor the inner loop into a method and use `return`.